packit_service/worker/helpers/testing_farm.py

```diff
@@ -290,6 +294,26 @@ def _payload_without_token(payload: Dict) -> Dict:
         payload_["notification"]["webhook"].pop("token")
         return payload_
 
+    def _construct_fmf_payload(self) -> dict:
+        fmf = {"url": self.fmf_url}
+        if self.fmf_ref:
+            fmf["ref"] = self.fmf_ref
+
+            # We assign a commit hash for merging only if:
+            # • there are no custom fmf tests set
+            # • we merge and have a PR
+            if (
+                not self.custom_fmf
+                and self.job_config.merge_pr_in_ci
+                and self.target_branch_sha
+            ):
+                fmf["merge_sha"] = self.target_branch_sha
```

Comment:
what are the expectations here?

I thought that we would send 2 hashes to TF so they know what to merge

Reply:
That's what's happening?

1º you can define `fmf_url` (and `fmf_ref`) → custom tests, no merging should happen, cause it's different repo
2º you don't have that defined, you must have enabled merging and present `target_branch_sha` (means it's not running for commit) → we pass `merge_sha`, so the TF can perform the merge

Reply:
ahaaa, so it's confusing naming then: I thought that merge_sha is the hash *after we merge*, not what's meant to be merged where

makes sense to me now